src/main/java/org/opensearch/security/ssl/transport/SPIFFEPrincipalExtractor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2015-2017 floragunn GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.opensearch.security.ssl.transport;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.SpecialPermission;
+
+public class SPIFFEPrincipalExtractor implements PrincipalExtractor {
+    /**
+     * PrincipalExtractor implementation that extracts a SPIFFE URI from an X.509 certificate's SAN.
+     * Returns "CN=spiffe://..." if found, otherwise null.
+     * Used for SPIFFE X.509 SVID-based authentication in OpenSearch clusters.
+     */
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+
+    @Override
+    @SuppressWarnings("removal")
+    public String extractPrincipal(final X509Certificate x509Certificate, final Type type) {
+        if (x509Certificate == null) {
+            return null;
+        }
+
+        final SecurityManager sm = System.getSecurityManager();
```

Comment:
Security manager is now removed so this will always return null. We can remove all JSM related code here.